sys/include/stdio_base.h

```diff
@@ -26,11 +26,65 @@
 #include <unistd.h>
 
 #include "modules.h"
+#include "isrpipe.h"
+#include "xfa.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+#ifndef STDIO_RX_BUFSIZE
+/**
+ * @brief Buffer size for STDIO
+ */
+#define STDIO_RX_BUFSIZE    (64)
+#endif
+
+enum {
+    STDIO_NULL,             /**< dummy stdio */
+    STDIO_UART,             /**< stdio via UART */
+    STDIO_RTT,              /**< stdio via Segger RTT */
+    STDIO_SEMIHOSTING,      /**< stdio via Semihosting */
+    STDIO_USBUS_CDC_ACM,    /**< stdio via USB CDC ACM (usbus) */
+    STDIO_TINYUSB_CDC_ACM,  /**< tdio via USB CDC ACM (TinyUSB) */
+    STDIO_ESP32_SERIAL_JTAG,/**< stdio via ESP32 debug Serial/JTAG */
+    STDIO_NIMBLE,           /**< stdio via BLE (NimBLE) */
+    STDIO_UDP,              /**< stdio via UDP */
+    STDIO_TELNET,           /**< stdio via telnet */
+    STDIO_ETHOS,            /**< stdio via ethos (mutiplex) */
+    STDIO_SLIP,             /*<< stdio via SLIP (mutiplex) */
+};
+
+/**
+ * @brief stdio provider struct
+ */
+typedef struct {
+    /**
+     * @brief   Initialize and attach the stdio provider
+     */
+    void (*attach)(void);
+    /**
+     * @brief   Detach the stdio provider
+     * @param[in] disable   Disable the stdio provider completely
+     */
+    void (*detach)(bool disable);
```

Comment:
Wouldn't this be `open`/`close` in classical file lingo? Why not use these names here?